src/io/dae.rs

```diff
@@ -0,0 +1,169 @@
+                            }
+                        }
+                        if let Some(n) = normal {
+                            if ((normals[ind * 3] - n.x as f32) as f32).abs() > std::f32::EPSILON
+                                || ((normals[ind * 3 + 1] - n.y as f32) as f32).abs()
+                                    > std::f32::EPSILON
+                                || ((normals[ind * 3 + 2] - n.z as f32) as f32).abs()
+                                    > std::f32::EPSILON
+                            {
+                                index = None;
+                            }
+                        }
+                    }
+
+                    if index.is_none() {
+                        index = Some(positions.len() / 3);
+                        map.insert(i.0, index.unwrap());
+                        let position = object.vertices[i.0];
+                        positions.push(position.x as f32);
+                        positions.push(position.y as f32);
+                        positions.push(position.z as f32);
+
+                        if let Some(tex) = uvw {
+                            uvs.push(tex.x as f32);
+                            uvs.push(tex.y as f32);
+                        }
+                        if let Some(n) = normal {
+                            normals.push(n.x as f32);
+                            normals.push(n.y as f32);
+                            normals.push(n.z as f32);
+                        }
+                    }
+
+                    indices.push(index.unwrap() as u32);
+                };
+
+                for shape in &geo.mesh[..] {
+                    match shape {
+                        collada::PrimitiveElement::Triangles(tris) => {
+                            for (i, tri_vert) in tris.vertices.to_vec().into_iter().enumerate() {
+                                let curr_tex_verts = match &tris.tex_vertices {
+                                    Some(tex_verts) => Some(tex_verts[i]),
+                                    None => None,
+                                };
+                                let curr_normals = match &tris.normals {
+                                    Some(norms) => Some(norms[i]),
+                                    None => None,
+                                };
+
+                                let dae_vtn0: DaeVTNIndex = (
+                                    tri_vert.0,
+                                    if let Some(v) = curr_tex_verts {
+                                        Some(v.0)
+                                    } else {
+                                        None
+                                    },
+                                    if let Some(n) = curr_normals {
+                                        Some(n.0)
+                                    } else {
+                                        None
+                                    },
+                                );
```

Comment:
Probably an easier way to deal with these three tuples

Reply:
maybe use map? Something like `curr_tex_verts.map(|v| v.0)`?